javagen/src/main/java/com/azure/autorest/model/clientmodel/ClientModel.java

```diff
@@ -204,26 +207,35 @@ public ImplementationDetails getImplementationDetails() {
 
     /**
      * List the properties that have access (getter or setter) methods.
-     *
-     * It does not include properties from superclass (even though they can be accessed via inheritance).
-     * It does not include properties that only have private access (e.g. property of a flattened model).
-     * It includes properties that can be accessed from the model but not declared in this model (e.g. properties from a flattened model).
+     * <p>
+     * It does not include properties from superclass (even though they can be accessed via inheritance). It does not
+     * include properties that only have private access (e.g. property of a flattened model). It includes properties
+     * that can be accessed from the model but not declared in this model (e.g. properties from a flattened model).
      *
      * @return The properties that have access (getter or setter) methods.
      */
     public List<ClientModelPropertyAccess> getAccessibleProperties() {
-        List<ClientModelProperty> properties1 = this.getProperties() == null ? Collections.emptyList() : this.getProperties();
-        List<ClientModelPropertyAccess> properties = properties1.stream()
-                .filter(p -> !p.getClientFlatten())
-                .collect(Collectors.toList());
-        properties.addAll(this.getPropertyReferences().stream()
-                .filter(ClientModelPropertyReference::isFromFlattenedProperty)
-                .collect(Collectors.toList()));
-        return properties;
+        List<ClientModelPropertyAccess> propertyAccesses = new ArrayList<>();
+        if (properties != null) {
+            for (ClientModelProperty property : properties) {
+                if (!property.getClientFlatten()) {
+                    propertyAccesses.add(property);
+                }
+            }
+        }
+
+        for (ClientModelPropertyReference clientModelPropertyReference : getPropertyReferences()) {
+            if (clientModelPropertyReference.isFromFlattenedProperty()) {
+                propertyAccesses.add(clientModelPropertyReference);
+            }
+        }
+
+        return propertyAccesses;
```

Comment:
nit, any concern over stream vs for-loop? Performance concern or what?

Usually, I view only it as personal preference?